src/core/qgsdatasourceuri.cpp

```diff
@@ -543,7 +543,7 @@ QString QgsDataSourceURI::uri( bool expandAuthConfig ) const
   if ( mWkbType != QGis::WKBUnknown && mWkbType != QGis::WKBNoGeometry )
   {
     theUri += " type=";
-    theUri += QgsWKBTypes::displayString( QgsWKBTypes::flatType(( QgsWKBTypes::Type( mWkbType ) ) ) );
+    theUri += QgsWKBTypes::displayString( QgsWKBTypes::Type( mWkbType ) );
```

Comment:
True.
I am thinking of keeping a new wkb type in the class and adding two methods newWkbType() and setNewWkbType(). And replace QGis::wkbType by QgsWKBTypes::Type in QGIS 3.0 ?